src/execution.jl

```diff
@@ -704,10 +712,18 @@ macro bprofile(args...)
     return esc(
         quote
             local $tmp = $BenchmarkTools.@benchmarkable $(args...)
-            $BenchmarkTools.warmup($tmp)
-            $(hasevals(params) ? :() : :($BenchmarkTools.tune!($tmp)))
+            $(
+                if hasevals(params)
```

Comment:
same remark on tuning

Reply:
In this case, hasevals is true, so we aren't calling tune, but we still want to warm up the code by running it once. We call 
`run($tmp, $BenchmarkTools.Parameters($tmp.params; evals=1); warmup=false)` to run one evaluation of the code, and we set warmup=false so as not to run it twice.

Reply:
In this case, hasevals is true so we don't need to tune, but we still need to warm up. We call run with evals=1 to run one execution of the code, and somewhat confusingly set warmup=false because we don't need to run it twice. The reason for all this fuss is we're trying to point the @profile macro at an already warmed up call to `run`